File: cses/sorting_and_searching/apartments.py
n, m, k = [int(x) for x in input().split(" ")]
a = [int(x) for x in input().split(" ")]
b = [int(x) for x in input().split(" ")]
a.sort()
b.sort()
ans, i, j = 0, 0 , 0
while i < n and j < m:
    x = a[i]
    if x - k <= b[j] <= x + k:
        ans += 1
        i += 1
        j += 1
    elif x < b[j]:
        i += 1
    elif x > b[j]:
        j += 1
print(ans)

File: cses/sorting_and_searching/Apartments.kt
fun main()  {
    val (n, m, k) = readLine()!!.split(" ").map{ it.toInt() }
    val a = readLine()!!.split(" ").map{ it.toInt() }.toIntArray()
    val b = readLine()!!.split(" ").map{ it.toInt() }.toIntArray()
    a.sort()
    b.sort()
    var ans = 0
    var i = 0
    var j = 0
    while (i < n && j < m)  {
        val x = a[i]
        if (x - k <= b[j] && b[j] <= x + k) {
            ans++
            i++
            j++
        }
        else if (x > b[j])
            j++
        else if (x < b[j])
            i++
    }
    println(ans)
}